File: May/May_19/count.go
package main

import (
	"fmt"
	"math/rand"
	"time"
)

func main() {
	rand.Seed(time.Now().UnixNano())
	numberToGuess := rand.Intn(100) + 1
	numberOfGuesses := 0
	var playerGuess int

	fmt.Println("Welcome to 'Guess the Number'!")
	fmt.Println("I'm thinking of a number between 1 and 100.")
	fmt.Println("Can you guess what it is?")

	for {
		fmt.Print("Enter your guess: ")
		_, err := fmt.Scan(&playerGuess)
		if err != nil {
			fmt.Println("Invalid input. Please enter a number.")
			continue
		}

		numberOfGuesses++

		if playerGuess > numberToGuess {
			fmt.Println("Too high! Try again.")
		} else if playerGuess < numberToGuess {
			fmt.Println("Too low! Try again.")
		} else {
			fmt.Printf("Congratulations! You've guessed the number in %d guesses.\n", numberOfGuesses)
			break
		}
	}
}

File: 5_May/May_18/count.cs
using System;

class GuessTheNumber
{
    static void Main(string[] args)
    {
        Random random = new Random();
        int numberToGuess = random.Next(1, 101);
        int numberOfGuesses = 0;
        bool hasGuessedCorrectly = false;

        Console.WriteLine("Welcome to 'Guess the Number'!");
        Console.WriteLine("I'm thinking of a number between 1 and 100.");
        Console.WriteLine("Can you guess what it is?");

        while (!hasGuessedCorrectly)
        {
            Console.Write("Enter your guess: ");
            string input = Console.ReadLine();
            int playerGuess;

            if (int.TryParse(input, out playerGuess))
            {
                numberOfGuesses++;

                if (playerGuess > numberToGuess)
                {
                    Console.WriteLine("Too high! Try again.");
                }
                else if (playerGuess < numberToGuess)
                {
                    Console.WriteLine("Too low! Try again.");
                }
                else
                {
                    Console.WriteLine($"Congratulations! You've guessed the number in {numberOfGuesses} guesses.");
                    hasGuessedCorrectly = true;
                }
            }
            else
            {
                Console.WriteLine("Invalid input. Please enter a number.");
            }
        }
    }
}
